lightning/src/chain/channelmonitor.rs

```diff
@@ -123,6 +123,26 @@ impl Readable for ChannelMonitorUpdate {
 #[derive(Clone, Debug)]
 pub struct MonitorUpdateError(pub &'static str);
 
+/// A [`MonitorEvent`] which indicates a [`ChannelMonitor`] update has completed. See
+/// [`ChannelMonitorUpdateErr::TemporaryFailure`] for more information on how this is used.
+///
+/// [`ChannelMonitorUpdateErr::TemporaryFailure`]: super::ChannelMonitorUpdateErr::TemporaryFailure
+#[derive(Clone, Debug, PartialEq)]
+pub struct MonitorUpdated {
```

Comment:
Not a huge fan of this name -- maybe `GlobalUpdateId` instead (since it's a unique ID across all the monitors)?

Reply:
Maybe `MonitorUpdatedEvent`? I kinda want to focus on the fact that its a part of an "event", though I agree the name kinda sucks.

Reply:
ACK `MonitorUpdatedEvent`

Reply:
Would be kinda strange to have a `MonitorEvent` enum contain a variant _holding_  `MonitorUpdatedEvent`, especially given `MonitorEvent` has a variant _named_ `HTLCEvent` (i.e. "event" is used inconsistently across three different parts).

Generally, I think using event in the variant names is redundant. Maybe name this struct `MonitorUpdateReference`? Another idea would be to give `monitor_update_id` a typedef and forgo the struct altogether as `UpdateCompleted(OutPoint, MonitorUpdateId)`.

Reply:
Is such a thing even possible? 🤯